File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    vector<int> storage = {1,2};
    int climbStairs(int n){
        if (n == 1){return 1;}
        if (n == 2){return 2;}
        while (n > 2){
            storage.push_back(storage[storage.size()-1]+storage[storage.size()-2]);
            n--;
        }
        return storage[storage.size()-1];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.swift
class Solution {
    func climbStairs(_ n: Int) -> Int {
        var dp = [0,1,2]
        if (n < 3){
            return dp[n]
        }
        for i in 3..<n+1{
            dp.append(dp[i-1]+dp[i-2])
        }
        return dp[n]   
    }
}